### PR TITLE
try to split at next key when split key is region start key

### PR DESCRIPTION
**Pull request by zhangjinpeng1987**
## What have you changed? (mandatory)
Try to split at next key when the current split key is region start key after adjusted. After batch split has imported, we will don't need this feature any more.

## What are the type of the changes? (mandatory)
Improvement

## How has this PR been tested? (mandatory)
Unit tests

## Does this PR affect documentation (docs/docs-cn) update? (mandatory)
No

## Does this PR affect tidb-ansible update? (mandatory)
No

### CODE REVIEW

**Comment by hicqu** [comment]
LGTM.

File: src/raftstore/coprocessor/split_observer.rs [hunk 1]
```diff
@@ -47,11 +47,19 @@ impl SplitObserver {
...
         }
 
-        let region_start_key = ctx.region().get_start_key();
+        let mut region_start_key = ctx.region().get_start_key();
 
-        let key = encode_bytes(&key);
+        let mut key = encode_bytes(&key);
         if *key <= *region_start_key {
-            return Err("no need to split".to_owned());
+            // Try next key
```
[the remark this answers is not in the document]
Comment (zhangjinpeng1987) [reply src/raftstore/coprocessor/split_observer.rs]: I guess < is used to tolerate the decode error before?

**Comment by siddontang** [comment]
PTAL @BusyJay

**Comment by BusyJay** [comment]
I'm worried about the small regions this pr will create, which can be a serious problem if merge is disabled. #3064 seems a better solution.

**Comment by zhangjinpeng1987** [comment]
> I'm worried about the small regions this pr will create, which can be a serious problem if merge is disabled. #3064 seems a better solution.

@BusyJay Please see the description `After batch split has imported, we will don't need this feature any more`.

**Comment by zhangjinpeng1987** [comment]
Batch split will cover this scenario and tikv-ctl support split a region by a specified key, so this pr can be abandoned.

**Pull request closed without merging**